### PR TITLE
Add frontend for no providers filtering

### LINKED ISSUE #266: The GitHub logo is out of place when data loads
### Before reporting

- [X] I have tested with the lastest _master_

### Describe the bug

The GitHub SVG is below the navbar while the data loads, then jumps into the right place

https://user-images.githubusercontent.com/43907402/179369240-13c3d344-4e08-4b05-8ceb-b996dbb8ec7b.mov

### To Reproduce

Catch at first load or spam f5 a bit :D 

### Expected behavior

Should be in its place while the data loads

### PR DESCRIPTION
**Pull request by Primdahl26**
# Description

Adds frontend for No providers filtering

Also refactors the filtering component to only use the session localstorage
Fixes #266 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: frontend/src/components/filters.svelte [hunk 1]
```diff
@@ -1,8 +1,7 @@
 <script lang="ts">
-  export let movieChecked = true
-  export let tvChecked = true
-  export let search
+  import { filters } from "../stores/filters.js"
 
+  export let search
```
Comment (AndreasPB) [inline frontend/src/components/filters.svelte:4]:
```suggestion
  export let search: string
```

Comment (Primdahl26) [reply frontend/src/components/filters.svelte]: Typed as a function `() => void`

Comment (AndreasPB) [reply frontend/src/components/filters.svelte]: Nvm it was a func, but we talked about it on Discord :D

File: frontend/src/components/filters.svelte [hunk 2]
```diff
@@ -87,8 +86,8 @@
           <input
             type="checkbox"
             class="checkbox"
-            bind:checked={movieChecked}
-            disabled={!tvChecked}
+            bind:checked={$filters.movieChecked}
+            disabled={!$filters.tvChecked}
             on:change={() => search()}
           />
         </label>
```
Comment (AndreasPB) [inline frontend/src/components/filters.svelte:93]: We are mixing checkboxes and toggles in the filter menu, it should probably just all be toggled for that modern look.

Also, both of the checkboxes in `Content type` goes gray when one is deselected - It should only be the one you shouldn't press.

File: frontend/src/components/no_results.svelte [hunk 3]
```diff
@@ -29,15 +29,19 @@
         <i>Consider adding more providers or removing all</i>
       </p>
     {/if}
-    {#if !$filters.tvChecked || !$filters.movieChecked}
-      <p class="pt-2">With filter:</p>
+    {#if !$filters.tvChecked || !$filters.movieChecked || !$filters.showNoProviders}
```
Comment (AndreasPB) [inline frontend/src/components/no_results.svelte:32]: A dynamic idea

```suggestion
    {#if Object.values($filters).includes(false)}
```

Comment (Primdahl26) [reply frontend/src/components/no_results.svelte:32]: We will need to make them all objects with a description for this to work. So either we make a issue and do it in another PR or it should happen here i guess.

**Review by AndreasPB — APPROVED** [review]

**Pull request merged**